Sprint 15. CoreData

shervik opened the pull request:
Реализован фреймворк CoreData

shervik committed:
add ColorCell to TrackerCreatingScreen

shervik committed:
draft coreData

shervik committed:
CoreData has been implemented

shervik committed:
мелкие исправления

shervik committed:
add keyboard hiding

DanilaG commented on Tracker/Presentation/TrackerCreatingScreen/CollectionView.swift:
```diff
@@ -9,41 +9,78 @@ import UIKit
+    func didEnabledCreateButton(isEnabledCreate: Bool, isEmptyWeekDay: Bool)
 }
 
 final class CollectionView: NSObject {
```
Давай поправим название класса, чтобы из него было видно зачем этот класс нужен или за что он отвечает. Так как название `CollectionView` слишком абстратное)

DanilaG commented on Tracker/Domain/CoreData/Store/TrackerRecordStore.swift:
```diff
@@ -0,0 +1,72 @@
+    func updateRecord(_ record: TrackerCoreData, date: Date) throws
+    func getCountRecords(for id: UUID) -> Int
+    func isTrackerCompleted(_ id: UUID, with date: Date) -> Bool
+}
```
Классно, что закрываешь классы протоколами. Это делает код более гибким и упрощает его тестирование)

DanilaG commented on Tracker/Presentation/TrackerCreatingScreen/SettingsCells/ColorCell.swift:
```diff
@@ -0,0 +1,49 @@
+import UIKit
+
+final class ColorCell: UICollectionViewCell {
+    static let identifier = "ColorCell"
```
Супер, что идентификатор в константе. Такой подход позволит снизить шансы опечаток и позволит пользоваться автодополнением

DanilaG commented on Tracker/Presentation/TrackerCreatingScreen/CollectionView.swift:
```diff
@@ -169,23 +206,84 @@ extension CollectionView: UICollectionViewDataSource {
+        case 1: if indexPath.row == 1 {
             scheduleVC = ScheduleViewController()
             scheduleVC?.delegate = self
             delegate?.didOpenScreen(scheduleVC ?? UIViewController())
```
Хорошо, что в МР-е нет ни одного force unwrap-a/cast-a. Это позволяет снизить шансы падения приложения)

shervik committed:
перенесено абстрагирование CoreData

перенесено абстрагирование CoreData из TrackerProvider в соответствующие классы TrackerStore, TrackerCategoryStore и TrackerRecordStore
